### PR TITLE
Add integration test for extra signers using tx preconditions

### PR DESCRIPTION
**Pull request by sreuland**
new integration test case for using the `Precondtions.ExtraSigners` parameter with a standard AccountID as the additional signer.

### CODE REVIEW

File: services/horizon/internal/integration/transaction_preconditions_test.go [hunk 1]
```diff
@@ -64,6 +64,38 @@ func TestTransactionPreconditionsTimeBounds(t *testing.T) {
...
+
+	// this errors because the tx preconditions require extra signer that
+	// didn't sign this tx
+	txParams.Preconditions.ExtraSigners = []string{addtlAccounts[0].GetAccountID()}
+	_, err = itest.SubmitMultiSigTransaction([]*keypair.Full{master}, txParams)
+	tt.Error(err)
+
+	// Now the transaction should be submitted without problems, the extra signer specified
+	// has also signed this transaction.
+	// TODO - doesn't work yet, need to figure out how
```
Comment (2opremio) [inline services/horizon/internal/integration/transaction_preconditions_test.go:93]: Leftover comment?

Comment (sreuland) [reply services/horizon/internal/integration/transaction_preconditions_test.go]: yep, thanks, removed.

File: services/horizon/internal/integration/transaction_preconditions_test.go [hunk 2]
```diff
@@ -64,6 +64,37 @@ func TestTransactionPreconditionsTimeBounds(t *testing.T) {
...
+	// didn't sign this tx
+	txParams.Preconditions.ExtraSigners = []string{addtlAccounts[0].GetAccountID()}
+	_, err = itest.SubmitMultiSigTransaction([]*keypair.Full{master}, txParams)
+	tt.Error(err)
+
+	// Now the transaction should be submitted without problems, the extra signer specified
+	// has also signed this transaction.
+	txParams.Preconditions.ExtraSigners = []string{addtlAccounts[0].GetAccountID()}
+	_, err = itest.SubmitMultiSigTransaction([]*keypair.Full{master, addtlSigners[0]}, txParams)
+	tt.NoError(err)
```
Comment (Shaptic) [inline services/horizon/internal/integration/transaction_preconditions_test.go:95]: We should also make sure that retrieving the tx by ID shows the extra signers in the response

Comment (sreuland) [reply services/horizon/internal/integration/transaction_preconditions_test.go:110]: great suggestion, [I added round trip asserts](https://github.com/stellar/go/pull/4320/commits/ded994277b13cb153cc734be2c352a8ed755f8f4) on the precond values in each of the test cases, thanks!

**Comment by 2opremio** [comment]
I just show that the verify-range test is failing, I tried to rerun it

**Comment by 2opremio** [comment]
@sreuland Is this PR ready otherwise? @Shaptic can you take a look if so?

**Review by Shaptic — APPROVED** [review]
Yep, LGTM!

**Pull request merged**